server/activity/activity.model.js

```diff
@@ -192,12 +192,18 @@ class Activity extends Model {
 
   reorder(index) {
     return this.sequelize.transaction(t => {
-      return this.siblings().then(siblings => {
+      const filter = this.getReorderFilter();
+      return this.siblings(filter).then(siblings => {
         this.position = calculatePosition(this.id, index, siblings);
         return this.save({ transaction: t });
       });
     });
   }
+
+  getReorderFilter() {
```

Comment:
Not sure if we need this as a separate method (since it won't be reused and is relatively simple).